prime-router/src/main/kotlin/history/azure/DeliveryFunction.kt

```diff
@@ -203,10 +209,108 @@ class DeliveryFunction(
         }
     }
 
+    /**
+     * Get a sortable list of facilities for multiple deliveries
+     *
+     * @param request HTML request body.
+     * @return JSON of the facility list or errors.
+     */
+    @FunctionName("getBulkDeliveryFacilities")
+    fun getBulkDeliveryFacilities(
+        @HttpTrigger(
+            name = "getBulkDeliveryFacilities",
+            methods = [HttpMethod.GET],
+            authLevel = AuthorizationLevel.ANONYMOUS,
+            route = "waters/report/{receiver}/facilities"
+        ) request: HttpRequestMessage<String?>,
+        @BindingName("receiver") receiver: String
+    ): HttpResponseMessage {
+        // Do authentication
+        val claims = AuthenticatedClaims.authenticate(request)
+            ?: return HttpUtilities.unauthorizedResponse(request, authenticationFailure)
+
+        val userOrgName = this.validateOrgSvcName(receiver)
+            ?: return HttpUtilities.notFoundResponse(
+                request,
+                "$receiver: invalid organization or service identifier"
+            )
+
+        // Authorize based on: org name in the path == org name in claim.  Or be a prime admin.
+        if (!claims.authorizedForSendOrReceive(userOrgName, null, request)) {
+            logger.warn(
+                "Invalid Authorization for user ${claims.userName}:" +
+                    " ${request.httpMethod}:${request.uri.path}." +
+                    " ERR: Claims scopes are ${claims.scopes} but client id is $userOrgName"
+            )
+            return HttpUtilities.unauthorizedResponse(request, authorizationFailure)
+        }
+        logger.info(
+            "Authorized request by org ${claims.scopes} to getList on organization $userOrgName."
+        )
+        try {
+            return run {
+                val pageSize = HistoryApiParameters(request.queryParameters).pageSize
+                val pageNumber = BulkFacilityListApiParameters(request.queryParameters).pageNumber
```

Comment:
Can we just parse the parameters once?

Reply:
We can't here since the parsing of `HistoryApiParameters` expects different parameters. The new class is using the params from the proposal doc instead